Source/SwiftLintFramework/Extensions/SwiftExpressionCallKind.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  SwiftExpressionCallKind.swift
+//  SwiftLint
+//
+//  Created by Marcelo Fabri on 11/12/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+
+public enum SwiftExpressionCallKind: String {
```

Comment:
I think we'd need to rethink our implementations if we did that. They only work because of the `other` case, as these kinds can be nested in other kinds (`struct` for example). If we added the other cases and removed the `other` one, some rules could start failing.

For example, `trailing_comma` in this example: 
```swift
struct Bar {
  let foo = [1: 2, 2: 3, ]
},
```

This enum also can't be `internal` because it's used as `associatedtype` on `ASTRule`s, in functions that are `public`.